deepchecks/utils/outliers.py

```diff
@@ -41,16 +41,35 @@ def iqr_outliers_range(data: np.ndarray,
     if len(iqr_range) != 2 or any((x < 0 or x > 100 for x in iqr_range)) or all(x < 1 for x in iqr_range):
         raise DeepchecksValueError('IQR range must contain two numbers between 0 to 100')
 
-    data = data.squeeze()
-    # Filter nulls
-    data = [x for x in data if pd.notnull(x)]
-    if len(data) < min_samples:
-        raise NotEnoughSamplesError(f'Need at least {min_samples} non-null samples to calculate IQR outliers, but got '
-                                    f'{len(data)}')
-
     q1, q3 = np.percentile(data, sorted(iqr_range))
     iqr = q3 - q1
     low_lim = q1 - scale * iqr
     up_lim = q3 + scale * iqr
 
     return low_lim, up_lim
+
+
+def sharp_drop_outliers_range(data_percents: Sequence, sharp_drop: float = 0.9,
+                              max_outlier_percentage: float = 0.05) -> Union[float, None]:
+    """Calculate outliers range on the data given using sharp drop.
+
+    Parameters
+    ----------
+    data_percents : np.ndarray
+        Counts of data to calculate outliers range for. The data is assumed to be sorted from the most common to the
+        least common.
+    sharp_drop : float , default 0.9
+        The sharp drop threshold to use for the outliers detection.
+    max_outlier_percentage : float , default 0.05
+        The maximum percentage of data that can be considered as "outliers".
+    """
+    if not 1 - EPS < sum(data_percents) < 1 + EPS:
+        raise DeepchecksValueError('Data percents must sum to 1')
+
+    for i in range(len(data_percents) - 1):
+        if sum(data_percents[:i+1]) < 1 - max_outlier_percentage:
+            continue
+        if 1 - (data_percents[i + 1] / data_percents[i]) >= sharp_drop:
```

Comment:
Theoretically this enables the following sequence:
50%, 4%, 3%, 3%... (add 3% untill 100%) in which case it's reasonable to say that there are no outliers, but the algorithm will return that 4% onwards are outliers. 

Way to avoid this is having a limit that the cumulative % of outliers can't be more than max_outlier_percentage

Reply:
@nirhutnik that's the main comment I'd like to discuss here

Reply:
it is aggregative :) Notice that I check sum(data_percents[:i+1]) , not data_percents[i+1]